tests/appsec/test_asm_standalone.py

```diff
@@ -0,0 +1,329 @@
+            assert span["metrics"]["_dd.apm.enabled"] == 0
+            assert "_dd.p.appsec" not in span["meta"]
+            assert "_dd.p.other" in span["meta"]
+            assert span["trace_id"] == 1212121212121212121
+
+            # Some tracers use true while others use yes
+            assert any(
+                ["Datadog-Client-Computed-Stats", trueish,] in data["request"]["headers"] for trueish in ["yes", "true"]
+            )
+
+    def setup_no_upstream_appsec_propagation__with_attack__is_kept_with_priority_2__from_minus_1(self):
+        trace_id = 1212121212121212121
+        parent_id = 34343434
+        self.r = weblog.get(
+            "/waf/",
+            headers={
+                "x-datadog-trace-id": str(trace_id),
+                "x-datadog-parent-id": str(parent_id),
+                "x-datadog-origin": "rum",
+                "x-datadog-sampling-priority": "-1",
+                "x-datadog-tags": "_dd.p.other=1",
+                "User-Agent": "Arachni/v1",
+            },
+        )
+
+    def test_no_upstream_appsec_propagation__with_attack__is_kept_with_priority_2__from_minus_1(self):
+        for data, _, span in interfaces.library.get_spans(request=self.r):
+            assert span["metrics"]["_sampling_priority_v1"] == 2
+            assert span["metrics"]["_dd.apm.enabled"] == 0
+            assert span["meta"]["_dd.p.appsec"] == "1"
+            assert span["trace_id"] == 1212121212121212121
+
+            # Some tracers use true while others use yes
+            assert any(
+                ["Datadog-Client-Computed-Stats", trueish,] in data["request"]["headers"] for trueish in ["yes", "true"]
+            )
+
+    def setup_no_upstream_appsec_propagation__with_attack__is_kept_with_priority_2__from_0(self):
+        trace_id = 1212121212121212121
+        parent_id = 34343434
+        self.r = weblog.get(
+            "/waf/",
+            headers={
+                "x-datadog-trace-id": str(trace_id),
+                "x-datadog-parent-id": str(parent_id),
+                "x-datadog-origin": "rum",
+                "x-datadog-sampling-priority": "0",
+                "x-datadog-tags": "_dd.p.other=1",
+                "User-Agent": "Arachni/v1",
+            },
+        )
+
+    def test_no_upstream_appsec_propagation__with_attack__is_kept_with_priority_2__from_0(self):
+        for data, _, span in interfaces.library.get_spans(request=self.r):
+            assert span["metrics"]["_sampling_priority_v1"] == 2
+            assert span["metrics"]["_dd.apm.enabled"] == 0
+            assert span["meta"]["_dd.p.appsec"] == "1"
+            assert span["trace_id"] == 1212121212121212121
+
+            # Some tracers use true while others use yes
+            assert any(
+                ["Datadog-Client-Computed-Stats", trueish,] in data["request"]["headers"] for trueish in ["yes", "true"]
+            )
+
+    def setup_upstream_appsec_propagation__no_attack__is_propagated_as_is__being_0(self):
```

Comment:
Is this real? I mean at least in java there is no chance to have an ASM event without force keep

Reply:
It shouldn't happen if every time you set the appsec propagation tag you also set the priority to force keep, but as far as I know we don't have tests for that yet, so I should have to believe you.

Otherwise you can set this test as irrelevant for java and problem solved.

Reply:
There is no real problem with the test, It should pass in java. :)